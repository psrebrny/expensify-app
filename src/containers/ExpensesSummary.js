import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

class ExpensesSummaryClass extends Component {
  
  render() {
    this.expenseWord = this.props.expensesCount <= 1 ? 'expense' : 'expenses';
    this.formattedExpensesTotal = numeral(this.props.expensesTotal).format('$0,0.00');
    
    return (
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">
              Viewing&nbsp;
              <span>{this.props.expensesCount}</span>&nbsp;
              {this.expenseWord} totalling&nbsp;
              <span>{this.formattedExpensesTotal}</span>
            </h1>
            <div className="page-header__actions">
              <Link className="button" to="/create">Add Expense</Link>
            </div>
          </div>
        </div>
    );
  }
  
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

const ExpensesSummary = connect(mapStateToProps, null)(ExpensesSummaryClass);

export { ExpensesSummaryClass, ExpensesSummary };