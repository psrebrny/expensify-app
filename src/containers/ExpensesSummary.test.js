import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummaryClass } from './ExpensesSummary';

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummaryClass expensesCount={1} expensesTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});


test('should correctly render expenses summary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummaryClass expensesCount={23} expensesTotal={2354355432534}/>);
  expect(wrapper).toMatchSnapshot();
});