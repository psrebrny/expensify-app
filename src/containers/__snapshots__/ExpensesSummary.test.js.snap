// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totalling 
      <span>
        $2,354,355,432,534.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $235.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
